Skip initial form for saved reports

gcsecsey opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/7615

## Proposed Changes

* skip the initial form for saved reports
* show the message  "Getting your report" on the loading screen

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* We're updating the Site Profiler UI

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to a saved report URL, eg. `/site-profiler/`
* Check that the initial form is skipped
* Check that the loading screen only displays the message "Getting your report..."
* Check with a new URL for regressions


https://github.com/Automattic/wp-calypso/assets/11555574/16b0acda-799b-4734-9e34-5cacd31b844e



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gcsecsey committed:
don't show check site form for saved results

gcsecsey committed:
update loading screen text for saved reports

gcsecsey committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into update/skip-form-for-saved-results

WBerredo reviewed (commented):
I'm not sure this is a regression, but when looked for a site in `/site-profiler` without the hash, it goes straight to the `Finalizing your report` message.
![CleanShot 2024-06-06 at 16 17 03](https://github.com/Automattic/wp-calypso/assets/5039531/58b9ae1e-d0e0-414f-a834-192010f56010)

epeicher commented:
HI @gcsecsey , I have merged https://github.com/Automattic/wp-calypso/pull/91575 and just remembered that I wanted to confirm with you first about this one. There is a merge conflict, but I think you can benefit from the changes in https://github.com/Automattic/wp-calypso/pull/91575 and probably apply the logic to the messages. 

Please let me know if any questions

gcsecsey committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into update/skip-form-for-saved-results

gcsecsey committed:
pass landing page referral as query parameter

gcsecsey commented:
> HI @gcsecsey , I have merged #91575 and just remembered that I wanted to confirm with you first about this one. There is a merge conflict, but I think you can benefit from the changes in #91575 and probably apply the logic to the messages.
> 
> Please let me know if any questions

Thanks for highlighting that @epeicher. I rebased and added a new query parameter to be able to identify the referrer, and to control the message displayed on the progressbar.

epeicher commented on client/site-profiler/components/site-profiler-v2.tsx:
```diff
@@ -39,10 +39,11 @@ import './styles-v2.scss';
 interface Props {
 	routerDomain?: string;
 	hash?: string;
+	routerRef?: string;
```
Non-blocking and very nit-pick 😛 but, the name of the `prop` seems to be related to a `React` ref (that's the first thing I thought). Do you think it could be changed to something like `routerOrigin` or another synonym?

gcsecsey replied:
Yep, that's a good idea!

epeicher reviewed (approved):
I have tested it, and it works as advertised. When the report is generated from the landing page, I can see all the sentences, but when coming from the Email, I only see the `Getting your report` as expected. LGTM!

gcsecsey committed:
rename `routerReef` to `routerOrigin`

epeicher reviewed (approved):
Thanks for addressing the comments! I have tested again and I see the expected results. LGTM! :shipit: 

|From Landing Page 1|From Landing Page 2 | From email|
|-------|------|------|
|      ![CleanShot 2024-06-07 at 16 03 39@2x](https://github.com/Automattic/wp-calypso/assets/3519124/f24759b1-5dd0-465b-954c-bf76d8faceb2)           |   ![CleanShot 2024-06-07 at 16 03 46@2x](https://github.com/Automattic/wp-calypso/assets/3519124/29b64861-670a-45b4-943c-fa380f2bc641)   |       ![CleanShot 2024-06-07 at 16 05 42@2x](https://github.com/Automattic/wp-calypso/assets/3519124/ef253534-d573-45b6-a604-ef1c2c9de290)          |

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14194327

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @gcsecsey, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Getting your report…`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14194327).
